src/back-end/lib/resources/affiliation.ts

```diff
@@ -58,63 +56,58 @@ const resource: Resource = {
 
   create(connection) {
     return {
-      async parseRequestBody(request) {
-        return request.body.tag === 'json' ? request.body.value : {};
+      async parseRequestBody(request): Promise<CreateRequestBody> {
+        const body = request.body.tag === 'json' ? request.body.value : {};
+        return {
+          user: getString(body, 'user'),
+          organization: getString(body, 'organization'),
+          membershipType: getString(body, 'membershipType')
+        };
       },
-      async validateRequestBody(request) {
+      async validateRequestBody(request): Promise<Validation<ValidatedCreateRequestBody, CreateValidationErrors>> {
```

Comment:
You shouldn't need to type the return value of this function, as it should be inferred by the `Resource` type on lines 30 & 31.